services/business/health.go

```diff
@@ -23,6 +29,75 @@ func (in *HealthService) GetServiceHealth(namespace, service, rateInterval strin
 	return health
 }
 
+// GetNamespaceHealth returns a health for all services in given Namespace (thus, it fetches data from K8S and Prometheus)
+func (in *HealthService) GetNamespaceHealth(namespace, rateInterval string) (NamespaceHealth, error) {
+	serviceList, err := in.k8s.GetServices(namespace)
+	if err != nil {
+		return nil, err
+	}
+	return in.getNamespaceHealth(namespace, serviceList, rateInterval), nil
+}
+
+func (in *HealthService) getNamespaceHealth(namespace string, sl *kubernetes.ServiceList, rateInterval string) NamespaceHealth {
+	allHealth := make(NamespaceHealth)
+
+	// Extract deployment statuses
+	for _, item := range sl.Services.Items {
+		sPods := kubernetes.FilterPodsForService(&item, sl.Pods)
+		depls := kubernetes.FilterDeploymentsForService(&item, sPods, sl.Deployments)
+		statuses := castDeploymentsStatuses(depls)
+		allHealth[item.Name] = &models.Health{
+			DeploymentStatuses: statuses,
+			DeploymentsFetched: true}
+	}
+
+	// Fetch services requests rates
```

Comment:
Perhaps the "Extract deployment statuses" and "Fetch services requests rates" blocks can be run concurrently

Reply:
Yeah but "Extract deployment statuses" is I think super-fast (probably a matter of few milliseconds if not nanos); it doesn't fetch k8s api, it already has everything in `sl *kubernetes.ServiceList`.